src/System.Text.Primitives/System/Text/Encoders/Utf8Decoder.cs

```diff
@@ -0,0 +1,264 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+namespace System.Text.Primitives.System.Text.Encoders
```

Comment:
That is a verbose namespace. Maybe System.Text.Primitives.Encoders? I thought we had a namespace with "Web" in the name somewhere.